js/views/modals/editListing/EditListing.js

```diff
@@ -863,6 +1078,66 @@ export default class extends BaseModal {
 
       this.$shippingOptionsWrap.append(shipOptsFrag);
 
+      // render sku field
+      if (this.skuField) this.skuField.remove();
+
+      this.skuField = this.createChild(SkuField, {
+        model: item,
+        initialState: {
+          variantsPresent: !!item.get('options').length,
+        },
+      });
+
+      this.$('.js-skuFieldContainer').html(this.skuField.render().el);
+
+      // render variants
+      if (this.variantsView) this.variantsView.remove();
+
+      this.variantsView = this.createChild(Variants, {
+        collection: this.variantOptionsCl,
+        maxVariantCount: item.max.optionCount,
+      });
+
+      this.variantsView.listenTo(this.variantsView, 'variantChoiceChange',
+        this.onVariantChoiceChange.bind(this));
+
+      this.$variantsSection.find('.js-variantsContainer').append(
+        this.variantsView.render().el
+      );
+
+      // render inventory management section
+      if (this.inventoryManagement) this.inventoryManagement.remove();
+      const inventoryManagementErrors = {};
+
+      if (this.model.validationError &&
+        this.model.validationError['item.quantity']) {
+        inventoryManagementErrors.quantity = this.model.validationError['item.quantity'];
+      }
+
+      this.inventoryManagement = this.createChild(InventoryManagement, {
+        initialState: {
+          trackBy: this.trackInventoryBy,
+          quantity: item.get('quantity'),
+          errors: inventoryManagementErrors,
+        },
+      });
+
+      this.$('.js-inventoryManagementSection').html(this.inventoryManagement.render().el);
+      this.listenTo(this.inventoryManagement, 'changeManagementType', this.onChangeManagementType);
+
+      // render variant inventory
+      if (this.variantInventory) this.variantInventory.remove();
+
+      this.variantInventory = this.createChild(VariantInventory, {
+        collection: item.get('skus'),
+        optionsCl: item.get('options'),
+        getPrice: () => this.getFormData(this.$itemPrice).item.price,
```

Comment:
If a user hasn't entered a price yet, this gets passed as a string with a value of zero instead of a number, which throws an error in the currency.js code. 

Casting it as a number fixed it when I was testing it.